guest/rust/examples/basics/procedural_generation/src/client.rs

```diff
@@ -0,0 +1,367 @@
+use ambient_api::client::{material, mesh, sampler, texture};
+use ambient_api::components::core::camera::aspect_ratio_from_window;
+use ambient_api::components::core::primitives::{cube, quad, sphere_radius};
+use ambient_api::components::core::procedurals::{procedural_material, procedural_mesh};
+use ambient_api::concepts::{
+    make_perspective_infinite_reverse_camera, make_sphere, make_transformable,
+};
+use ambient_api::prelude::*;
+use noise::{utils::*, Fbm, Perlin};
+use palette::IntoColor;
+
+const TAU: f32 = std::f32::consts::TAU;
+const RESOLUTION_X: u32 = 32;
+const RESOLUTION_Y: u32 = 8;
+const TEXTURE_RESOLUTION_X: u32 = 4 * RESOLUTION_X;
+const TEXTURE_RESOLUTION_Y: u32 = 4 * RESOLUTION_Y;
+const SIZE_X: f32 = RESOLUTION_X as f32 / RESOLUTION_Y as f32;
+const SIZE_Y: f32 = 1.0;
+const WAVE_AMPLITUDE: f32 = 0.25;
+const WAVE_FREQUENCY: f32 = 0.5 * TAU;
+const ROTATING_SUN: bool = false;
+
+fn make_camera() {
+    Entity::new()
+        .with_merge(make_perspective_infinite_reverse_camera())
+        .with(aspect_ratio_from_window(), EntityId::resources())
+        .with_default(main_scene())
+        .with(translation(), vec3(0.0, 3.0, 4.0) * 2.0)
+        .with(lookat_target(), vec3(0.0, 3.0, 0.0))
+        .spawn();
+}
+
+fn make_lighting() {
+    Entity::new()
+        .with_merge(make_transformable())
+        .with_default(sun())
+        .with(
+            rotation(),
+            Quat::from_rotation_y(-45_f32.to_radians())
+                * Quat::from_rotation_z(-45_f32.to_radians()),
+        )
+        .with(light_diffuse(), Vec3::ONE * 4.0)
+        .with_default(main_scene())
+        .spawn();
+    if ROTATING_SUN {
+        query(rotation()).requires(sun()).each_frame(move |suns| {
+            for (sun_id, sun_rotation) in suns {
+                entity::set_component(
+                    sun_id,
+                    rotation(),
+                    Quat::from_rotation_z(frametime()) * sun_rotation,
+                );
+            }
+        });
+    }
+}
+
+fn make_coordinate_system() {
+    let make_cube = |t: Vec3, s: Vec3, c: Vec4| {
```

Comment:
style nitpick: if these don't close over any style, prefer `fn make_cube(...)` to make that clearer